vendor/README.md

```diff
@@ -0,0 +1,319 @@
+Vendor tests
+============
+
+When creating a badge for a new service or changing a badge's behavior, tests
+should be included. They serve three purposes:
+
+1. The contributor and revewer can easily verify the code works as
+   intended.
+
+2. When a badge stops working on the live server, maintainers can find out
+   right away.
+
+3. They speed up future contributors when they are debugging or improving a
+   badge.
+
+Contributors should take care to cover each part of a badge's functionality,
+and ideally, all code branches:
+
+- Typical case
+- Customized cases
+- Server errors
+- Not found errors
+- Parse errors
+
+
+Tutorial
+--------
+
+In this tutorial, we'll write tests for the Travis badge:
+
+```js
+camp.route(/^\/travis(-ci)?\/([^\/]+\/[^\/]+)(?:\/(.+))?\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  var userRepo = match[2];  // eg, espadrine/sc
+  var branch = match[3];
+  var format = match[4];
+  var options = {
+    method: 'HEAD',
+    uri: 'https://api.travis-ci.org/' + userRepo + '.svg',
+  };
+  if (branch != null) {
+    options.uri += '?branch=' + branch;
+  }
+  var badgeData = getBadgeData('build', data);
+  request(options, function(err, res) {
+    if (err != null) {
+      console.error('Travis error: ' + err.stack);
+      if (res) { console.error(''+res); }
+      badgeData.text[1] = 'invalid';
+      sendBadge(format, badgeData);
+      return;
+    }
+    try {
+      var state = res.headers['content-disposition']
+                     .match(/filename="(.+)\.svg"/)[1];
+      badgeData.text[1] = state;
+      if (state === 'passing') {
+        badgeData.colorscheme = 'brightgreen';
+      } else if (state === 'failing') {
+        badgeData.colorscheme = 'red';
+      } else {
+        badgeData.text[1] = state;
+      }
+      sendBadge(format, badgeData);
+
+    } catch(e) {
+      badgeData.text[1] = 'invalid';
+      sendBadge(format, badgeData);
+    }
+  });
+}));
+```
+
+Before getting started, install the project dependencies if you haven't
+already:
+
+```
+npm i
+```
+
+We'll start by creating a new module, `vendor/travis.js`, using this
+boilerplate:
+
+```js
+'use strict';
+
+const Joi = require('joi');
+const ServiceTester = require('./runner/service-tester');
+
+const t = new ServiceTester('Travis', '/travis');
+module.exports = t;
+```
+
+We'll add all our tests to this ServiceTester object, which gets exported from
+the module. The two arguments to the constructor are the name of the service
+and its URI prefix. The tester will prepend it to the URIs you provide later,
+which saves copying and pasting.
+
+Next we'll add a test for the typical case.
+
+[![](https://img.shields.io/badge/build-passing-brightgreen.svg)]()
+
+The JSON format for this badge is `{ name: 'build', value: 'passing' }`.
+
+Here's what our first test looks like:
+
+```js
+t.create('build status on default branch')
+  .get('/rust-lang/rust.json')
```

Comment:
I understand! https://github.com/badges/shields/pull/937/files#diff-2bcb895f83ee1e2e5fc386962bf9d6c0R157
This is the url we want to  test.